cmd/prow-job-dispatcher/main.go

```diff
@@ -370,4 +424,61 @@ func main() {
 	if err := dispatcher.SaveConfig(config, o.configPath); err != nil {
 		logrus.WithError(err).Fatalf("Failed to save config file to %s", o.configPath)
 	}
+
+	if !o.createPR {
+		logrus.Info("Finished dispatching and create no PR, exiting ...")
+		os.Exit(0)
+	}
+
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
+		logrus.WithError(err).Fatal("Failed to change to root dir")
+	}
+
+	steps := []struct {
+		gitOnly   bool
+		command   string
+		arguments []string
+	}{
+		{
+			gitOnly:   true,
```

Comment:
So `gitOnly` means you define a command and arguments and then never call it? That is confusing at best. I'd rather have it you make a `commandWithArgs ...string` variadic arg to `runAndCommitIfNeeded` and do not call anything (aka what is called "git only" today) `if len(commandWithArgs == 0)`